### PR TITLE
Introduce UuidParseError

### LINKED ISSUE #276: Refactor ParseError
Current `ParseError` assumes that only `Hyphenated`, `Simple` and `Urn` + `SIV 128` exist out in the universe. We need to address this so people can have their own parser for other formats (like URL blob uuid) and can provide their own context for the error.

### PR DESCRIPTION
**Pull request by kinggoesgaming**
**I'm submitting a(n)** feature,refactor

# Description
Replace ParseError with UuidParseError

# Motivation
Currently, the context for the errors around uuid string is limited to hyphenated, integer, simple and urn formats. This expands the error layout so that the context is more generic so nonstandard formats can use our error type for custom uuids.

# Tests
Fix current tests to match the new signature. No new tests added

# Related Issue(s)
closes #276

### CODE REVIEW

File: src/parser/core_support.rs [hunk 1]
```diff
@@ -12,7 +12,7 @@
 use core::fmt;
 use parser;
 
-impl<'a> fmt::Display for parser::Expected<'a> {
+impl<'a> fmt::Display for parser::Expected {
```
Comment (Dylan-DPC) [inline src/parser/core_support.rs]: you don't need the `<'a>` on impl

Comment (kinggoesgaming) [reply src/parser/core_support.rs]: oops

**Comment by kinggoesgaming** [comment]
@Dylan-DPC ready for review + merge

**Comment by Dylan-DPC** [comment]
bors: r+

**Pull request merged**